### PR TITLE
Various links and buttons now use new colour variables

### PR DESCRIPTION
**Pull request by PeculiarGoat**
ADO card: ☑️ [AB#5270](https://dev.azure.com/S72/SHIFT72/_workitems/edit/5270)

Elab notes/AC: 📃 [Google Docs](https://docs.google.com/document/d/1sAErB8qOx-vDhmh0YD2JbCrRI7puqZs92qLbYxmmbxA/edit#)
- [x] Has this been discussed with Delivery Team?

Designs: 
- [Figma](https://www.figma.com/file/0vuK8I8PA78KgNv5BzY8qU/Primary-Colour-Injection?node-id=1%3A562)

## Description of work
- CSS variables improvements:
  - Improved notes on how to calculate various colours.
  - Added body colour hover variable (for use with body colour elements that have a hover state).
  - Added hover background variable (for use with elements that require a background colour around them on hover).
  - Added new variables for nav links, search, dropdowns, signin button, and signout button, and removed the corresponding Sass variables as they are now redundant.
- Signin button text now dims on hover.
- CTA icon buttons now dim on hover and have a background circle on hover.
- Cookie consent link now removes underline on hover.
- Footer links now dim on hover.
- Powered-by link now removes underline on hover.
- Account modal cancel buttons now match the signout button styles.
- Modal X buttons now dim on hover.
- Cast and crew links now remove underline on hover.
- Links on pages now dim on hover.
- Nav search now dims on hover, and bottom-border is primary colour when open.
- Dropdowns and carets now dim on hover.
- Dropdown items now have a button background on hover.  Margins and padding have been adjusted to accomodate this.
- Nav links now dim on hover.
- Share modal items now have a button background on hover.  Fixed some dodgy styles to accomodate this.

## Caveats
- The Figma design shows are circle background on hover for the search icon.  I left this out of the PR because it wont work without an overhaul of the sketchy search CSS.

## Checklist
- [x] CI tests are passing Github actions (inc. linting)
- [x] Key areas of the feature outlined for context and testing
- [x] If there are designs for this work are they noted here and in the ADO card 
- [x] Design review
- [x] Have checked this at multiple screen resolutions and range of browsers
- [x] Moved ADO card to Dev/done, checked link to Github, tagged "Review"
- [x] Updated changelog (if applicable)
- [x] I promise to document any new feature toggles/configurations in the appropriate documentation

### CODE REVIEW

**Review by curtisadams28 — CHANGES_REQUESTED** [review]
image
The sign out button is still the primary colour on hover.

File: site/styles/_variables.scss [hunk 1]
```diff
@@ -4,36 +4,88 @@
...
+    //   3.  Convert the result back to RGB.
+    //   TODO(jackh): figure out a non-Sass way of calculating this.
+    --input-focus-border-color-rgb: 131, 210, 216;
+    
+    // To calculate --hover-background-rgb, do:
+    //   1.  Convert var(--body-bg-accent-rgb) to hex.
+    //   2a. If dark themed, run it through the Sass lighten function:  lighten(hex, 13%).
+    //   2b. If light themed, run it through the Sass darken function:  darken(hex, 13%).
+    //   3.  Convert the result back to RGB.
+    //   TODO(jackh): figure out a non-Sass way of calculating this.
```
Comment (smitsanghvi21) [inline site/styles/_variables.scss:43]: Might need to tidy up some comments

**Review by l0ud0gg — APPROVED** [review]
bar the other comments, looks good. ran it up and looked at all the changes quickly in chrome / macos 👍

File: site/styles/_variables.scss (hunk 1 above)
Comment (smitsanghvi21) [reply site/styles/_variables.scss]: Looks like ` TODO(jackh)` comments are still there at couple of places. Do you want to keep it there lol?

Comment (PeculiarGoat) [reply site/styles/_variables.scss]: Yeah they are TODOs...

Comment (smitsanghvi21) [reply site/styles/_variables.scss]: Looks bit weird if there are TODO comments with a name on the main branch as this is a public repo.

Comment (PeculiarGoat) [reply site/styles/_variables.scss]: There's nothing unusual about having TODOs and the name is so people know who to talk to about it if they need to.

**Comment by curtisadams28** [comment]
@PeculiarGoat The translations dropdown is overflowing.
image

**Review by curtisadams28 — APPROVED** [review]
LGTM 👍

**Pull request merged**